src/output.cpp

```diff
@@ -497,21 +502,38 @@ void Output::map_stats_contents(
 
     auto key_str = map_key_to_str(bpftrace, map, key);
 
-    int64_t count = (int64_t)value.at(0);
-    int64_t total = value.at(1);
-    int64_t average = 0;
-
-    if (count != 0)
-      average = total / count;
+    std::string total_str;
+    std::string count_str;
+    std::string avg_str;
+
+    if (map_type.IsSigned()) {
+      int64_t total = 0;
+      int64_t count = 0;
+      int64_t avg = stats_value<int64_t>(value, bpftrace.ncpus_, total, count);
+      avg_str = std::to_string(map_type.IsStatsTy() ? avg : avg / div);
+      total_str = std::to_string(total);
+      count_str = std::to_string(count);
+    } else {
+      uint64_t total = 0;
+      uint64_t count = 0;
+      uint64_t avg = stats_value<uint64_t>(
+          value, bpftrace.ncpus_, total, count);
+      avg_str = std::to_string(map_type.IsStatsTy() ? avg : avg / div);
```

Comment:
Not your code (and I'm not suggesting you change it), but why on earth do we have to deal with this conditional division during output?? It should have been sorted out way earlier (probably semantic analysis) so that these later stages don't have to care 😞

Reply:
We probably don't as div should always be `1` for the stats type and you're right semantic analyser should catch this. I'll remove.